README.md

```diff
@@ -1,2 +1,237 @@
+
+The format of the configuration file is shown in the following example:
+
+```
+# Template for repo-tools configuration
+name: <Repository name>
+description: <Repository description>
+version: <Repository version>
+stacks:
+  - name: <Repository index name>
+    repos:
+      - url: <Reference to index file>
+      exclude:
+          - <stack name>
+      include:
+          - <stack name>
+image-org: <Organization containing images within registry>
+image-registry: <Image registry hosting images>
+nginx-image-name: <Image name for the generated nginx image, defaults to repo-index>
+```
+
+
+The options shown in the configuration file are described in the following list:
+
+- `name:` An identifier for this particular configuration.
+- `description:` A unique description for your configuration.
+- `version:` A version number, which might align with a release. For example, a Git release of your stack hub.
+- `stacks:`
+    - `name:` The name of a repository to build, which might contain one or more application stacks.
+    - `repos:` An array of URLs that point to one or more stack index files that you want to include in this repository
+        - `url:` `exclude:` (Optional) An array of stack names that you want to exclude from the repository. If no filtering is required, leave blank.
+        - `url:` `include:` (Optional) An array of stack names that you want to include from the repository. If no filtering is required, leave blank.
+- `image-org:` (Optional) The name of the organization within the image registry where you want to store the docker images for included stacks. See **Configuring public or private repositories**.
+- `image-registry:` (Optional) The URL of the image registry where you want to store the stack docker images. See **Configuring public or private repositories**.
+- `nginx-image-name:` (Optional) The name assigned to the generated NGINX image. The default is `repo-index`. See **Configuring public or private repositories**.
+
+NOTE: The `exclude:` and `include:` fields are mutually exclusive. Use only one or the other for a specific repository to avoid build errors.
+
+**Configuring public or private repositories**
+
+If your stacks and repositories are publicly available, leave the `image-org` and `image-registry` fields blank.
+
+If your stacks and repositories are hosted in a private environment that your deployment environment and tools cannot access, such as GitHub Enterprise, the build can create an NGINX image. This image can serve the assets you need from within your deployment environment without the need for authentication. Use the `image-org` and `image-registry` fields to define the organization and URL where you store your registry images. You can optionally configure a name for your NGINX image, which you must push to the registry that your deployment environment can access. When deployed, the NGINX image serves the repository index files that you create as part of the build.
```

Comment:
The NGINX image serves all the resources for each stack as well as the index files.

Reply:
Updated